Patch for rand 0.7 based on a partial revert of upstream commit
f54a37cf2a9e109c280688276de672cf1c7d5591

Index: time/Cargo.toml
===================================================================
--- time.orig/Cargo.toml
+++ time/Cargo.toml
@@ -42,7 +42,7 @@ version = "0.4.7"
 optional = true
 
 [dependencies.rand]
-version = "0.8.4"
+version = "0.7.3"
 optional = true
 default-features = false
 
@@ -52,7 +52,7 @@ optional = true
 default-features = false
 
 [dev-dependencies.rand]
-version = "0.8.4"
+version = "0.7.3"
 default-features = false
 
 [dev-dependencies.serde]
Index: time/src/rand.rs
===================================================================
--- time.orig/src/rand.rs
+++ time/src/rand.rs
@@ -8,10 +8,10 @@ use crate::{Date, Duration, Month, Offse
 impl Distribution<Time> for Standard {
     fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Time {
         Time::__from_hms_nanos_unchecked(
-            rng.gen_range(0..24),
-            rng.gen_range(0..60),
-            rng.gen_range(0..60),
-            rng.gen_range(0..1_000_000_000),
+            rng.gen_range(0, 24),
+            rng.gen_range(0, 60),
+            rng.gen_range(0, 60),
+            rng.gen_range(0, 1_000_000_000),
         )
     }
 }
@@ -19,14 +19,14 @@ impl Distribution<Time> for Standard {
 impl Distribution<Date> for Standard {
     fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Date {
         Date::from_julian_day_unchecked(
-            rng.gen_range(Date::MIN.to_julian_day()..=Date::MAX.to_julian_day()),
+            rng.gen_range(Date::MIN.to_julian_day(), Date::MAX.to_julian_day() + 1),
         )
     }
 }
 
 impl Distribution<UtcOffset> for Standard {
     fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> UtcOffset {
-        let seconds = rng.gen_range(-86399..=86399);
+        let seconds = rng.gen_range(-86399, 86399+1);
         UtcOffset::__from_hms_unchecked(
             (seconds / 3600) as _,
             ((seconds % 3600) / 60) as _,
@@ -51,7 +51,7 @@ impl Distribution<OffsetDateTime> for St
 impl Distribution<Duration> for Standard {
     fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Duration {
         Duration::nanoseconds_i128(
-            rng.gen_range(Duration::MIN.whole_nanoseconds()..=Duration::MAX.whole_nanoseconds()),
+            rng.gen_range(Duration::MIN.whole_nanoseconds(), Duration::MAX.whole_nanoseconds()+1),
         )
     }
 }
@@ -60,7 +60,7 @@ impl Distribution<Weekday> for Standard
     fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Weekday {
         use Weekday::*;
 
-        match rng.gen_range(0..7) {
+        match rng.gen_range(0, 7) {
             0 => Monday,
             1 => Tuesday,
             2 => Wednesday,
@@ -75,7 +75,7 @@ impl Distribution<Weekday> for Standard
 impl Distribution<Month> for Standard {
     fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Month {
         use Month::*;
-        match rng.gen_range(1..=12) {
+        match rng.gen_range(1,12+1) {
             1 => January,
             2 => February,
             3 => March,
